cpu/nrf52/radio/nrf802154/nrf802154_radio.c

```diff
@@ -714,6 +714,17 @@ static int _set_csma_params(ieee802154_dev_t *dev, const ieee802154_csma_be_t *b
     return 0;
 }
 
+void nrf802154_setup(nrf802154_t *dev)
+{
+    (void) dev;
+#if IS_USED(MODULE_NETDEV_IEEE802154_SUBMAC)
+    netdev_t *netdev = (netdev_t*) dev;
+    netdev_register(netdev, NETDEV_NRF802154, 0);
+    netdev_ieee802154_submac_init(&dev->netdev, &nrf802154_hal_dev);
```

Comment:
You'll need to call `netdev_ieee802154_setup()` after register to configure the MAC address.